code/system/application/perception/include/detectlane/detectlane.hpp

```diff
@@ -56,34 +58,50 @@ class DetectLane
  private:
   void setUp();
   void tearDown();
-  odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode body();
+  // odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode body();
 
   bool m_initialized;
-  cv::Mat m_image;
+  cv::Mat m_currentImg;
+  uint16_t m_blurKernelSize; 
+  cv::Mat m_cannyImg;
+  cv::Mat m_adapThreshImg;
   std::deque<std::pair<odcore::data::TimeStamp, cv::Mat>> m_visualMemory;
-  uint16_t m_intensityThreshold;
+  uint8_t m_adapThreshKernelSize;
+  uint8_t m_adapThreshConst;
   uint16_t m_cannyThreshold;
   uint16_t m_houghThreshold;
+  std::vector<cv::Vec2f> m_linesRaw;
+  std::vector<cv::Vec2f> m_linesProcessed;
+  std::vector<uint16_t> m_laneLineIds;
+  std::vector<uint16_t> m_currentLaneLineIds;
+  std::vector<cv::Vec2f> m_xScreenP;
+  std::vector<cv::Vec2f> m_yScreenP;
+  std::vector<cv::Vec2f> m_xWorldP;
+  std::vector<cv::Vec2f> m_yWorldP;
+  float m_lineDiff;
+  float m_OneLineDiff;
+  float m_HorisontalLimit;
   double m_memThreshold;
   double m_upperLaneLimit;
   double m_lowerLaneLimit;
   int16_t m_screenSize[2];
   int16_t m_roi[4];
   odcore::base::Mutex m_mtx;
   bool m_debug;
+  std::string m_cameraName; 
   Eigen::Matrix3d m_transformationMatrix;
 
```

Comment:
There are very many global variables here. Is it possible to group some of them in new classes and then point at such objects with unique_ptr? This could also be convenient when the data is returned from a method.